sympy/vector/implicitregion.py

```diff
@@ -0,0 +1,64 @@
+from sympy import tan, pi
+from sympy.abc import theta
+from sympy.core import expand_trig, Eq
+from sympy.solvers import solve
+from sympy.simplify.trigsimp import trigsimp
+from sympy.vector import ParametricRegion
+
+
+def ImplicitConicSection(equation, x, y):
+    """
+    Returns the parametric form of the conic section defined using implict equation.
+
+    Example
+    =======
+
+    >>> from sympy.abc import x, y
+    >>> from sympy.vector import ImplicitConicSection, vector_integrate
+
+    >>> ImplicitConicSection(Eq(x**2 + y**2 -1), x, y)
+    ParametricRegion((sin(2*theta), -cos(2*theta)), (theta, 0, pi))
+
+    >>> c = ImplicitConicSection((x - 3)**2 + (y + 2)**2 - 16, x, y)
+    >>> c
+    ParametricRegion((2*(-sqrt(7)*tan(theta) + 3)*cos(theta)**2, 3*sin(2*theta) + sqrt(7)*cos(2*theta) - 2), (theta, 0, pi))
+    >>> vector_integrate(1, c)
+    8*pi
+
+    >>> ellipse = ImplicitConicSection(x**2/4 + y**2/16 - 1, x, y)
+    >>> ellipse
+    ParametricRegion((8*tan(theta)/(tan(theta)**2 + 4), -4 + 8*tan(theta)**2/(tan(theta)**2 + 4)), (theta, 0, pi))
+    >>> vector_integrate
+    ### It gets stuck
+
+    """
+
+    if isinstance(equation, Eq):
+        equation = equation.lhs - equation.rhs
+
+    # Finding  a point on the Curve.
+    # Needs to find a better way than random looping.
+    for i in range(100):
```

Comment:
I need to find a better way to find a point on the conic.

Reply:
have you thought about some deterministic-iterative algorithm?

Reply:
If you cannot solve the equation... maybe you are at a singular point, right? Can you determine singular points from the gradient of the implicit equation? Maybe thinking of example equations where this case happens, may help think about a solution.

Reply:
I am working on it. I think I will implement a general algorithm for monoids which will work for all conics.

Reply:
Found on an algorithm for determining a point on the conic: 
https://www3.risc.jku.at/publications/download/risc_1355/Rational%20Points%20on%20Conics.pdf.

@Upabjojr the algorithm is non-trivial. Is it worth it to implement? Even parametrizing conics seems to be a tough task,

Reply:
The only approach which is working for me is to iterate through some points and determine whether it lies on the curve. Although an ugly approach, this should work for most cases.

Reply:
Another option is to make the user input a point on the curve apart from implicit equation.

Reply:
>  Is it worth it to implement? Even parametrizing conics seems to be a tough task,

How much time do you think this would take? If you ask me for a criterion on how to proceed... I would suggest to maximize the amount of "useful" stuff to add to SymPy. By "useful" you have to consider how many people may also be interested in using it. If lots of people would find this algorithm useful, maybe it's worth investing time on it.

Reply:
Maybe also worth considering: what would you implement as an alternative to that algorithm?

Reply:
> Another option is to make the user input a point on the curve apart from implicit equation.

An example:
```python
C = ImplicitCurve(Eq(x**2 + y**2 - 4))
C.parametrization(Point(0, 2)) #Takes a point on a curve
(2*cos(theta), 2*sin(theta))
```

@Upabjojr Any thoughts?

Reply:
Suggesting the point is fine with me, especially if it makes the code simpler to implement.

Reply:
We can work with this and maybe later implement the algorithm for finding a rational point on the curve.

Reply:
Of course... always start with a simple algorithm. We can open as many pull requests as we like, so no need to do it now.

Reply:
Thanks. I will complete this work then. This point will be optional. If not given, the algorithm will iterate over some point. If it is unable to find a point on the curve, it raises NotImplementedError.